### ISSUE TITLE
Bootstrapping files are created in wrong directory.

### ISSUE THREAD

**Issue by bgrube** [dglazier/brufit#40]
It seems that https://github.com/dglazier/brufit/blob/e342c984c4e7d4343fe479d0198fec113dc292bd/core/Data.cpp#L56 sets the wrong directory for those bootstrap files that belong to kinematic bins. This can be seen when running `tutorials/sPlotEventsPDF/FitPeakHistBinsBoot.C`: The script defines bins in the `Eg` variable and `Binner` creates the directories
```
 Eg3.100000_
 Eg3.300000_
 Eg3.500000_
 Eg3.700000_
 Eg3.900000_
```
However, the `Boot*.root` files for the `Eg` bins are not created in these directories but in the parent directory.

As a consequence, https://github.com/dglazier/brufit/blob/e342c984c4e7d4343fe479d0198fec113dc292bd/core/Data.cpp#L60 overwrites the `Boot*.root` for the lower `Eg` bins and only the ones for `Eg3.900000_` remain.

Unfortunately fixing this is not straight-forward. `Binner` would need an accessor to get the bin directory and this info would need to be passed down to `FitManager::fData` in `FitManager::LoadData()` and `FitManager::ReloadData()` and maybe other places as well.

**Comment by dglazier** [comment]
Whoops! Thanks for catching that. The fix was actually much simpler, the tree to be bootstrapped knows which directory it is from, and so it should be used to place the boot trees in the correct directory. Actually the correct line was there commented out! I am not sure why I changed it, just rushed a bit.
https://github.com/dglazier/brufit/blob/boot/core/BootStrapper.cpp#L58
Note I have merged this change into moms2amps too.

**Comment by bgrube** [comment]
Thanks a lot, Derek, for fixing this so quickly! The fits for the kinematic bins now work correctly. However, I'm afraid a special case is needed for fits, where no kinematic binning is used. In this case the current code writes the `Boot*.root` files into the directory of the input data. You can see this by commenting out the `RF.Bins().LoadBinVar("Eg",5,3,4);` line in `FitPeakHistBins.C` and `FitPeakHistBinsBoot.C`. If you run both scripts, the `Boot*.root` files are created in the `tutorials/sPlotEventsPDF` directory instead of `tutorials/sPlotEventsPDF/outPeakHistBinsBoot`. So the output directory for the bootstrap files depends on whether `Binner` is used or not.

**Comment by dglazier** [comment]
Thanks for testing and pointing this out, I probably made the original change because of that. I have added a check to make sure the outdir is in the desitination path as a way around this.

**Comment by bgrube** [comment]
Perfect! Thanks, Derek! The bootstrapping now works with and without binning. This issue was fixed by e1b7dfd0c7ed6ef2e33c623b4c4cd0225a13f1aa and e17bd6c45869741520ea61c986c2ea6def403c5a.
